staging/src/k8s.io/client-go/testing/fixture.go

```diff
@@ -296,6 +329,9 @@ func (t *tracker) add(gvr schema.GroupVersionResource, obj runtime.Object, ns st
 		}
 		if oldMeta.GetNamespace() == newMeta.GetNamespace() && oldMeta.GetName() == newMeta.GetName() {
 			if replaceExisting {
+				for _, w := range t.getWatches(gvr, ns) {
+					go w.Modify(obj)
```

Comment:
I would rather use a buffer than thread these out. The random order of reception would cause issues.